pkg/stats/metadata.go

```diff
@@ -39,28 +39,57 @@ func NewMetadata(ctx context.Context, logger logging.Logger, blockstoreType stri
 		}
 		res.Entries = append(res.Entries, MetadataEntry{Name: k, Value: v})
 	}
-	if cloudMetadataProvider != nil {
-		cloudMetadata := cloudMetadataProvider.GetMetadata()
-		for k, v := range cloudMetadata {
-			res.Entries = append(res.Entries, MetadataEntry{Name: k, Value: v})
+
+	for _, id := range storageConfig.GetStorageIDs() {
+		if adapterCfg := storageConfig.GetStorageByID(id); adapterCfg != nil {
+			appendOrUpdateEntry(res, BlockstoreTypeKey, adapterCfg.BlockstoreType())
+
+			provider := buildMetadataProvider(logger, adapterCfg)
+			cloudMetadata := provider.GetMetadata()
+			if cloudMetadata == nil {
+				cloudMetadata = getErrorMetadata()
+			}
+			for k, v := range cloudMetadata {
+				appendOrUpdateEntry(res, k, v)
+			}
 		}
 	}
-	res.Entries = append(res.Entries, MetadataEntry{Name: BlockstoreTypeKey, Value: blockstoreType})
+
 	return res
 }
 
+func appendOrUpdateEntry(metadata *Metadata, key, value string) {
+	for i, entry := range metadata.Entries {
+		if entry.Name == key {
+			metadata.Entries[i].Value = entry.Value + "," + value
+			return
+		}
+	}
+	metadata.Entries = append(metadata.Entries, MetadataEntry{Name: key, Value: value})
+}
+
+func getErrorMetadata() map[string]string {
+	return map[string]string{
+		cloud.IDKey:     "err",
+		cloud.IDTypeKey: "err",
+	}
+}
+
```

Comment:
What's ErrorMetadata? Is it stats we want to collect?

Reply:
In some cases we know the blockstore type, but failing to get the metadata from the cloud-provider.
In this case, we'll report `err` to the BI (instead of `nil`).

Reply:
Was this requested?

Reply:
Not as a product requirement, but from a functional pov - 
You can check the tests here, and see that once it's `nil` and not `err`, the BI consumer will think it's not configured, instead of this being configured but failing.

I can return `nil` here, but this also means we won't be able to unit-test these params.